include/mqtt/async_client.hpp

```diff
@@ -46,7 +59,20 @@ class async_client : public client<Socket, PacketIdBytes> {
      * @param port port number
      * @return async_client object
      */
-    friend std::shared_ptr<callable_overlay<async_client<tcp_endpoint<as::ip::tcp::socket, null_strand>>>>
+    friend std::shared_ptr<
+        callable_overlay<
+            async_client<
+                tcp_endpoint<
+                    as::ip::tcp::socket,
+#if defined(MQTT_NO_TS_EXECUTORS)
+                    as::io_context::executor_type
+#else
+                    null_strand
+#endif
```

Comment:
```suggestion
                    MQTT_NULL_STRAND
```

And int the header file strand.hpp, add the following macro.


```cpp
// Include and header comment like other header files....

#if defined(MQTT_NO_TS_EXECUTORS)
#define MQTT_NULL_STRAND  as::io_context::executor_type
#else  // defined(MQTT_NO_TS_EXECUTORS)
#define MQTT_NULL_STRAND null_strand
#endif
```

Reply:
That makes perfect sense.
Would you like me to define MQTT_STRAND and MQTT_NULL_STRAND both in `strand.hpp`?
Or should MQTT_NULL_STRAND be defined in `null_strand.hpp`?
```c++
#if !defined(MQTT_NULL_STRAND_HPP)
#define MQTT_NULL_STRAND_HPP

#if defined(MQTT_NO_TS_EXECUTORS)

#define MQTT_NULL_STRAND as::io_context::executor_type

#else // defined(MQTT_NO_TS_EXECUTORS)

// null_strand code as is

#define MQTT_NULL_STRAND null_strand

#endif // defined(MQTT_NO_TS_EXECUTORS)

#endif // MQTT_NULL_STRAND_HPP
```

It seems awkward to
```c++
#define MQTT_NULL_STRAND null_strand
```
in `strand.hpp` but also have a conditional include in `client.hpp` and `server.hpp`.
```c++
#if !defined(MQTT_NO_TS_EXECUTORS)
#include <mqtt/null_strand.hpp>
#endif
```

Please share your thoughts.